android/src/main/kotlin/es/antonborri/home_widget/HomeWidgetPlugin.kt

```diff
@@ -101,8 +101,8 @@ class HomeWidgetPlugin : FlutterPlugin, MethodCallHandler, ActivityAware,
                 }
             }
             "registerBackgroundCallback" -> {
-                val dispatcher = (call.arguments as Iterable<*>).toList()[0] as Long
-                val callback = (call.arguments as Iterable<*>).toList()[1] as Long
+                val dispatcher = (call.arguments as Iterable<*>).toList()[0].toString().toLong()
```

Comment:
Converting this `toString` feels really hacky. I guess it would also work if you cast the argument with `as Number` and call `toLong` on that

Reply:
Yes, I agree this is a less hacky solution, I implemented it and it works this way too. I fixed to casting `as Number` and then calling `toLong()`